tests/transforms/test_experimental/test_transform_program.py

```diff
@@ -107,6 +108,27 @@ def postprocessing2(results):
         out2 = _apply_postprocessing_stack(results, [postprocessing2, postprocessing1])
         assert out2 == (4.0, 9.0)
 
+    def test_prune_dynamic_transform(self):
+        """Tests that prune dynamic transform works."""
+
+        program1 = TransformProgram(
+            [
+                qml.transforms.dynamic_one_shot,
+                qml.transforms.sum_expand,
+                qml.transforms.dynamic_one_shot,
+            ]
+        )
+        program2 = TransformProgram(
+            [
+                qml.transforms.dynamic_one_shot,
+                qml.transforms.sum_expand,
+            ]
+        )
+
```

Comment:
Could include a test for the priority between `mid_circuit_measurements` and `dynamic_one_shot`? :thinking:

Reply:
Done